src/main/scala/chisel3/util/GrayCode.scala

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util
+
+import chisel3._
+
+object BinaryToGray {
+
+  /** Turns a binary number into gray code. */
+  def apply(in: UInt): UInt = in ^ (in >> 1.U)
```

Comment:
```suggestion
  def apply(in: UInt): UInt = in ^ (in >> 1)
```
This should be marginally more efficient in complication speed